### ISSUE TITLE
This CODE have a bug

### ISSUE THREAD

**Issue by GZR4619** [foryichuanqi/Remaining-useful-life-prediction-by-TaFCN#1]
ValueError: Found unexpected instance while processing input tensors for keras functional model. Expecting KerasTensor which is from tf.keras.Input() or output from keras layer call(). Got: 0

How to solve this BUG?
I tried a lot of things,but I don't solved this BUG

**Comment by foryichuanqi** [comment]
Sorry for this problem.
I want to tell you the following.

(1) I downloaded and ran the uploaded code. The interpretability_analysis.py you pointed out runs normally and gets the expected results.
(2) Based on (1) and your error, I think it may be caused by the version problems of keras and TF. Our versions are keras 2.2.4 and tensorflow gpu 1.15.0.
(3) If you have any questions, please do not hesitate to contact me and provide more details.

**Comment by CJLUhandsomeman** [comment]
I have the same problem as above, my tensorflow and keras versions are the same as yours, is it because of my 3060 graphics card and 11.6 CUDA?

**Comment by foryichuanqi** [comment]
> 

I don't think it is the cuda version. It may be caused by the Tensorflow version. The official version corresponding to Keras2.2.4 is Tensorflow 1.13.0. You can try to reduce the version to 1.13.0. However, this involves the relationship between the cuda version supported by Tensorflow 1.13.0 and the cuda version supported by your GTX3060. I suggest that if you try to do this, you can clone your original environment first, and then operate in the cloned environment. In this way, even if there is no success, it will not damage your original environment. If you can provide detailed error information, maybe I can provide more help.

**Comment by CJLUhandsomeman** [comment]
Thank you very much for taking the time out of your busy schedule to reply to me. I will debug it again. If possible, can you add your contact information, because I am very concerned about your content.


810912143
***@***.***



&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "foryichuanqi/Remaining-useful-life-prediction-by-TaFCN"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2022年10月25日(星期二) 晚上9:34
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [foryichuanqi/Remaining-useful-life-prediction-by-TaFCN] This CODE have a bug (Issue #1)





   
I don't think it is the cuda version. It may be caused by the Tensorflow version. The official version corresponding to Keras2.2.4 is Tensorflow 1.13.0. You can try to reduce the version to 1.13.0. However, this involves the relationship between the cuda version supported by Tensorflow 1.13.0 and the cuda version supported by your GTX3060. I suggest that if you try to do this, you can clone your original environment first, and then operate in the cloned environment. In this way, even if there is no success, it will not damage your original environment. If you can provide detailed error information, maybe I can provide more help.

**Comment by foryichuanqi** [comment]
You're welcome. I'm glad to be here to help.

> Thank you very much for taking the time out of your busy schedule to reply to me. I will debug it again. If possible, can you add your contact information, because I am very concerned about your content. 810912143 ***@***.*** &nbsp;
> […](#)
> ------------------&nbsp;原始邮件&nbsp;------------------ 发件人: "foryichuanqi/Remaining-useful-life-prediction-by-TaFCN" ***@***.***&gt;; 发送时间:&nbsp;2022年10月25日(星期二) 晚上9:34 ***@***.***&gt;; ***@***.******@***.***&gt;; 主题:&nbsp;Re: [foryichuanqi/Remaining-useful-life-prediction-by-TaFCN] This CODE have a bug (Issue #1) I don't think it is the cuda version. It may be caused by the Tensorflow version. The official version corresponding to Keras2.2.4 is Tensorflow 1.13.0. You can try to reduce the version to 1.13.0. However, this involves the relationship between the cuda version supported by Tensorflow 1.13.0 and the cuda version supported by your GTX3060. I suggest that if you try to do this, you can clone your original environment first, and then operate in the cloned environment. In this way, even if there is no success, it will not damage your original environment. If you can provide detailed error information, maybe I can provide more help. — Reply to this email directly, view it on GitHub, or unsubscribe. You are receiving this because you commented.Message ID: ***@***.***&gt;



> Thank you very much for taking the time out of your busy schedule to reply to me. I will debug it again. If possible, can you add your contact information, because I am very concerned about your content. 810912143 ***@***.*** &nbsp;
> […](#)
> ------------------&nbsp;原始邮件&nbsp;------------------ 发件人: "foryichuanqi/Remaining-useful-life-prediction-by-TaFCN" ***@***.***&gt;; 发送时间:&nbsp;2022年10月25日(星期二) 晚上9:34 ***@***.***&gt;; ***@***.******@***.***&gt;; 主题:&nbsp;Re: [foryichuanqi/Remaining-useful-life-prediction-by-TaFCN] This CODE have a bug (Issue #1) I don't think it is the cuda version. It may be caused by the Tensorflow version. The official version corresponding to Keras2.2.4 is Tensorflow 1.13.0. You can try to reduce the version to 1.13.0. However, this involves the relationship between the cuda version supported by Tensorflow 1.13.0 and the cuda version supported by your GTX3060. I suggest that if you try to do this, you can clone your original environment first, and then operate in the cloned environment. In this way, even if there is no success, it will not damage your original environment. If you can provide detailed error information, maybe I can provide more help. — Reply to this email directly, view it on GitHub, or unsubscribe. You are receiving this because you commented.Message ID: ***@***.***&gt;

**Comment by Mekako** [comment]
Is there any other way to be compatible with TensorFlow besides downgrading it?
除了降级TensorFlow没有别的办法能兼容吗，降级+换cuda环境还是挺麻烦的。

**Comment by foryichuanqi** [comment]
> Is there any other way to be compatible with TensorFlow besides downgrading it? 除了降级TensorFlow没有别的办法能兼容吗，降级+换cuda环境还是挺麻烦的。

I suggest that you don't use the model I gave you, and use my code to train a new model for interpretability analysis.我建议你不要使用我给的模型，用我的代码训练一个新模型，用于可解释性分析。
